espnet/nets/pytorch_backend/transformer/decoder.py

```diff
@@ -195,20 +195,45 @@ def __init__(
                 for lnum in range(num_blocks)
             ]
 
```

Comment:
Please add some comments about the difference (e.g., speech, speech_mask, return_hidden. What are they doing?).